src/llvm/di.rs

```diff
@@ -617,44 +149,52 @@ impl DISanitizer {
                                                 // BTF maps accepted by the Linux kernel.
                                                 if base_type_name == "AyaBtfMapMarker" {
                                                     // Remove the name from the struct.
-                                                    di_composite_type
-                                                        .replace_name(self.context, "")
-                                                        .unwrap();
+                                                    remove_name = true;
                                                     // And don't include the field in the sanitized DI.
                                                 } else {
-                                                    members.push(di_derived_type.di_type);
+                                                    members.push(di_derived_type.into());
                                                 }
                                             } else {
-                                                members.push(di_derived_type.di_type);
+                                                members.push(di_derived_type.into());
                                             }
                                         }
                                         _ => {
-                                            members.push(di_derived_type.di_type);
+                                            members.push(di_derived_type.into());
                                         }
                                     }
                                 }
                                 _ => {}
                             }
                         }
-                        if is_data_carrying_enum {
-                            di_composite_type.replace_elements(MDNode::empty(self.context));
-                        } else if !members.is_empty() {
+                        if !members.is_empty() {
```

Comment:
Here we do this work, but if is_data_carrying_enum then we discard the members right? So we can avoid doing this if is_data_carrying_enum (like it was before the diff I think)

Reply:
Right, done